src/Microsoft.AspNetCore.Razor.Test.Sources/CaseSensitiveTagHelperAttributeComparer.cs

```diff
@@ -0,0 +1,64 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using Microsoft.AspNetCore.Html;
+using Microsoft.AspNetCore.Razor.TagHelpers;
+
+namespace Microsoft.AspNetCore.Razor.Test.Internal
```

Comment:
This is an ugly namespace. We almost-always avoid `.Test` segments and only use `.Internal` in test class namespaces when the subject `class` is in the same `.Internal` namespace. (Well, both MVC and Razor violate the coding guidelines and use `.Test` more than they should.)

In any case, this is a test utility so I understand the `.Test` portion but the namespace should be much closer to `TagHelperAttribute`'s. Suggest `Microsoft.AspNetCore.Razor.TagHelpers.Test` or &hellip;`.Testing`.